vernac/record.mli

```diff
@@ -36,6 +36,50 @@ val definition_structure
   -> Ast.t list
   -> GlobRef.t list
 
+  module Data : sig
+    type projection_flags = {
+      pf_subclass: bool;
+      pf_canonical: bool;
+    }
+    type raw_data
+    type t =
+      { id : Id.t
+      ; idbuild : Id.t
+      ; is_coercion : bool
+      ; coers : projection_flags list
+      ; rdata : raw_data
+      ; inhabitant_id : Id.t
+      }
+  end
+
+  module Record_decl : sig
```

Comment:
Would be good to put a comment here mentioning a todo to further refactor this with the code for inductives, which I believe it can be done.

Reply:
I suppose that the goal of this PR is not really to refactor the vernac for records to much as it was to expose API that Enrico wished to use. When investigating how to go about https://github.com/coq/coq/issues/14737 I ran into a few problems, mainly that it is not entirely clear what the data of a record actually is. Many functions require only partial bits of the record data, so it is not so obvious how to organize this. I don't think that will be solved in this PR, so I second putting a comment for the future here.